server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -1493,13 +1505,19 @@ private void maybeFinish() {
                         originalListener.onFailure(e);
                         return;
                     }
-                    originalListener.onResponse(response);
+                    try {
+                        originalListener.onResponse(response);
+                    } finally {
+                        releaseResponse(response);
```

Comment:
`createFinalResponse()` (above on line 1491/1503) is what calls `getMergedResponse` on the `SearchResponseMerger`, so for now that one is getting decRef'd - does it also need to be incRef'd?

Reply:
We start out with reference count `1` above in the `getMergedResponse` and pass the response to the listener. The listener should be reference count neutral, i.e. take care of release any references it acquires itself down the road. For us the response goes out of scope here so we need to `decRef` the single reference we acquired by virtue of creating the response.